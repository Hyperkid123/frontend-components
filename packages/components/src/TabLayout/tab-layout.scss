.ins-tab-layout {
  > .ins-tabs {

    div {
      display: inline-block;
      margin-right: 30px;
      text-align: center;
      cursor: pointer;
      color: var(--pf-v5-global--link--Color);

      &.active {
        border-bottom: 3px var(--pf-v5-global--link--Color) solid;
      }

      &:hover {
        color: var(--pf-v5-global--link--Color--hover);
        border-bottom: 3px var(--pf-v5-global--link--Color--hover) solid;
      }
    }
  }
}
