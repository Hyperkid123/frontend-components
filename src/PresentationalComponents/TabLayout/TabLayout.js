import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

const TabLayout = ({ children, items, classNames, active, onTabClick, ...props }) => (
    <section { ...props } className={ classnames(classNames, 'ins-tab-layout') } widget-type='InsightsTabs'>
        <div className="ins-tabs">
            { items.map(oneItem => (
                <div key={ oneItem.name }
                    className={ classnames({ active: oneItem.name === active }) }
                    onClick={ event => onTabClick && onTabClick(event, oneItem) }
                    widget-type='InsightsTabsItem'
                    widget-id={ oneItem.name }
                >
                    { oneItem.title }
                </div>
            )) }
        </div>
        <div className="ins-tab-content">
            { children }
        </div>
    </section>
);

TabLayout.propTypes = {
    items: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string,
        title: PropTypes.string
    })),
    children: PropTypes.node,
    classNames: PropTypes.string,
    active: PropTypes.string,
    onTabClick: PropTypes.func
};

TabLayout.defaultProps = {};

export default TabLayout;
