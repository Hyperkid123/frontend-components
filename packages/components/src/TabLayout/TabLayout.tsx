import React from 'react';
import classnames from 'classnames';
import './tab-layout.scss';

export type TabLayoutItem = { name?: string; title?: React.ReactNode };

export interface TabLayoutProps {
  items?: TabLayoutItem[];
  classNames?: string;
  /**
   * Active tab item
   */
  active?: string;
  onTabClick?: (event: React.MouseEvent<HTMLDivElement>, oneItem: TabLayoutItem) => void;
}

/**
 * @deprecated
 *
 * Please use PF tabs component instead
 *
 */
const TabLayout: React.FunctionComponent<React.PropsWithChildren<TabLayoutProps>> = ({
  children,
  items = [],
  classNames,
  active,
  onTabClick = () => undefined,
  ...props
}) => (
  <section {...props} className={classnames(classNames, 'ins-tab-layout')} widget-type="InsightsTabs">
    <div className="ins-tabs">
      {items.map((oneItem) => (
        <div
          key={oneItem.name}
          className={classnames({ active: oneItem.name === active })}
          onClick={(event) => onTabClick(event, oneItem)}
          widget-type="InsightsTabsItem"
          widget-id={oneItem.name}
        >
          {oneItem.title}
        </div>
      ))}
    </div>
    <div className="ins-tab-content">{children}</div>
  </section>
);

export default TabLayout;
