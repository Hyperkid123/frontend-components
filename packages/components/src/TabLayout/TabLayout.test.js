import React from 'react';
import TabLayout from './TabLayout';
import { render } from '@testing-library/react';

const items = [
  {
    name: 'one',
    title: <div>one</div>,
  },
  {
    name: 'two',
    title: 'two',
  },
];

describe('TabLayout component', () => {
  describe('should render', () => {
    it('without items', () => {
      const { container } = render(<TabLayout />);
      expect(container).toMatchSnapshot();
    });

    it('with items', () => {
      const { container } = render(<TabLayout items={items} />);
      expect(container).toMatchSnapshot();
    });

    it('with active item', () => {
      const { container } = render(<TabLayout items={items} active="one" />);
      expect(container).toMatchSnapshot();
    });
  });
});
