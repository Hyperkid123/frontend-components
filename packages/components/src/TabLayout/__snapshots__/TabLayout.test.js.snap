// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabLayout component should render with active item 1`] = `
<div>
  <section
    class="ins-tab-layout"
    widget-type="InsightsTabs"
  >
    <div
      class="ins-tabs"
    >
      <div
        class="active"
        widget-id="one"
        widget-type="InsightsTabsItem"
      >
        <div>
          one
        </div>
      </div>
      <div
        class=""
        widget-id="two"
        widget-type="InsightsTabsItem"
      >
        two
      </div>
    </div>
    <div
      class="ins-tab-content"
    />
  </section>
</div>
`;

exports[`TabLayout component should render with items 1`] = `
<div>
  <section
    class="ins-tab-layout"
    widget-type="InsightsTabs"
  >
    <div
      class="ins-tabs"
    >
      <div
        class=""
        widget-id="one"
        widget-type="InsightsTabsItem"
      >
        <div>
          one
        </div>
      </div>
      <div
        class=""
        widget-id="two"
        widget-type="InsightsTabsItem"
      >
        two
      </div>
    </div>
    <div
      class="ins-tab-content"
    />
  </section>
</div>
`;

exports[`TabLayout component should render without items 1`] = `
<div>
  <section
    class="ins-tab-layout"
    widget-type="InsightsTabs"
  >
    <div
      class="ins-tabs"
    />
    <div
      class="ins-tab-content"
    />
  </section>
</div>
`;
